test/services/sync/strategies/consumerActions/ClassAction.test.js

```diff
@@ -0,0 +1,78 @@
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
+
+const sinon = require('sinon');
+const { ObjectId } = require('mongoose').Types;
+
+const { BadRequest } = require('../../../../../src/errors');
+const { SyncError } = require('../../../../../src/errors/applicationErrors');
+const { ClassAction } = require('../../../../../src/services/sync/strategies/consumerActions');
+
+const { SchoolRepo, ClassRepo } = require('../../../../../src/services/sync/repo');
+
+const { expect } = chai;
+chai.use(chaiAsPromised);
+
+describe('Class Actions', () => {
+	let classAction;
+
+	before(async () => {
+		classAction = new ClassAction(true);
```

Comment:
With the repo injections like it is added before we can pass the different fake repos as second parameter. But it also include the key list at the moment, but for test it is ok. (and by this way we must move the new Class to each it)

But again in the end both way are ok, but we should commit us to one. Keep in mind why we do it and which pros and cons it includes.

This is a part of a documentation ( where i really like if they is as part of the action as readme.md :P ) on confluence, where we write "We do no dependencie injection for repos because ...."